plan/plan.go

```diff
@@ -40,6 +41,10 @@ type Plan interface {
 	// If index can be used, a new index plan is returned, 'filtered' is true.
 	// If no index can be used, the original plan is returned and 'filtered' return false.
 	Filter(ctx context.Context, expr expression.Expression) (p Plan, filtered bool, err error)
+
+	// Next returns the next row of data and rowKeys, nil data means there is no more data to return.
+	// Aggregation plan will fetch all the data at the first call.
+	Next(ctx context.Context) (row *Row, err error)
```

Comment:
How about when user terminate, we check if the plan implements `io.Closer`, if it does implement it, we call `Close`?

And we can also extract `Filter` method to `Filterer` interface, so we don't need every plan to implement it.